content_script.js

```diff
@@ -104,14 +104,29 @@
           clipboardItems['text/rtf'] = new Blob(
             [
               `{\field{\*\fldinst HYPERLINK "${url}"}{\fldrslt ${
-                  selectedText}}}`,
+                  selection.toString()}}}`,
             ],
             {
               type: 'text/rtf',
             }
           );
           */
         }
+        else if (linkStyle === 'rich_plus_raw') {
+          var html = "";
+          if (selection.rangeCount) {
+              var container = document.createElement("div");
+              for (var i = 0, len = selection.rangeCount; i < len; ++i) {
+                  container.appendChild(selection.getRangeAt(i).cloneContents());
+              }
+              html = container.innerHTML;
+          }
+          clipboardItems['text/html'] = new Blob(
+              [`${html} <a href="${encodeURI(url)}">🔗</a>`],
```

Comment:
```suggestion
              [`${html} 🔗 <a href="${encodeURI(url)}">${encodeURI(url)}</a>`],
```

I think this is what you meant?!

Reply:
Unfortunately this change is not what I wanted. To take my use case for instance, I copy a few sentences or a whole paragraph sometimes and the URL could get very long in this case, and it looks concise if shown as a link of the emoji. 

Example:

I want to see:

Democrats have so far used the hearings to praise Brown -- who would be the first Black woman to serve as a Supreme Court justice -- as an exceptionally qualified, trail-blazing nominee whose depth and breadth of experience, including as a federal public defender, would add a valuable and unique perspective to the bench. [🔗](https://www.cnn.com/2022/03/22/politics/ketanji-brown-jackson-confirmation-hearings-day-two/index.html#:~:text=Democrats%20have%20so,to%20the%20bench.)

instead of

Democrats have so far used the hearings to praise Brown -- who would be the first Black woman to serve as a Supreme Court justice -- as an exceptionally qualified, trail-blazing nominee whose depth and breadth of experience, including as a federal public defender, would add a valuable and unique perspective to the bench.  🔗 [https://www.cnn.com/2022/03/22/politics/ketanji-brown-jackson-confirmation-hearings-day-two/index.html#:~:text=Democrats%20have%20so,to%20the%20bench.]( [🔗](https://www.cnn.com/2022/03/22/politics/ketanji-brown-jackson-confirmation-hearings-day-two/index.html#:~:text=Democrats%20have%20so,to%20the%20bench.))

Reply:
Another example to show that the text itself is with HTML style:

Want to get and see:

President Joe Biden's Supreme Court nominee, [Ketanji Brown Jackson, ](https://www.cnn.com/2022/02/02/politics/ketanji-brown-jackson-profile/index.html)defended her judicial record Tuesday as she faced intense questioning from Republican senators during the second day of her historic confirmation hearings. [🔗](https://www.cnn.com/2022/03/22/politics/ketanji-brown-jackson-confirmation-hearings-day-two/index.html#:~:text=president%20joe%20biden's%20supreme%20court%20nominee%2C%20ketanji%20brown%20jackson%2C%20defended%20her%20judicial%20record%20tuesday%20as%20she%20faced%20intense%20questioning%20from%20republican%20senators%20during%20the%20second%20day%20of%20her%20historic%20confirmation%20hearings.)

instead of

President Joe Biden's Supreme Court nominee, [Ketanji Brown Jackson, ](https://www.cnn.com/2022/02/02/politics/ketanji-brown-jackson-profile/index.html)defended her judicial record Tuesday as she faced intense questioning from Republican senators during the second day of her historic confirmation hearings. 🔗 [https://www.cnn.com/2022/03/22/politics/ketanji-brown-jackson-confirmation-hearings-day-two/index.html#:~:text=president%20joe%20biden's%20supreme%20court%20nominee%2C%20ketanji%20brown%20jackson%2C%20defended%20her%20judicial%20record%20tuesday%20as%20she%20faced%20intense%20questioning%20from%20republican%20senators%20during%20the%20second%20day%20of%20her%20historic%20confirmation%20hearings.](https://www.cnn.com/2022/03/22/politics/ketanji-brown-jackson-confirmation-hearings-day-two/index.html#:~:text=president%20joe%20biden's%20supreme%20court%20nominee%2C%20ketanji%20brown%20jackson%2C%20defended%20her%20judicial%20record%20tuesday%20as%20she%20faced%20intense%20questioning%20from%20republican%20senators%20during%20the%20second%20day%20of%20her%20historic%20confirmation%20hearings.)

Reply:
Ha this is what I was thinking. But the options could be very personal choices like emoji, Link, LINK, or link?

How about 1) 🔗 2) [source] 3) [link]? I prefer to use the square brackets to discriminate it from the text and use all lowercases.

Reply:
Think about international users as well. Thinking about it, I would ideally allow the user to choose:

- Quotes and quote style 
- Parentheses and parentheses style  around link word
- Link word

Two examples:

- “Bla bla” [Source]
- «Bla bla» (#)

Reply:
Good points! Quotes around the text looks nice for some use cases, but it doesn't make sense to me when the HTML text include multiple lines, which I usually copy. I would try to add the option to let user input any preferred "link text", and put the URL behind it.

Reply:
Link text options are added in the setting page to let the users either choose a preset or input their own favorite word. e6e83376abcc6cda6f5070c101f832605994213a @tomayac Please have a try!